lib/block-supports/background.php

```diff
@@ -35,20 +35,10 @@ function gutenberg_register_background_support( $block_type ) {
  *
  * @since 6.6.0
  *
- * @param  array  $background_styles Background style properties.
+ * @param  array $background_styles Background style properties.
  * @return array                     Style engine array of CSS string and style declarations.
  */
 function gutenberg_get_background_support_styles( $background_styles = array() ) {
-	$background_image_source             = isset( $background_styles['backgroundImage']['source'] ) ? $background_styles['backgroundImage']['source'] : null;
-	$background_styles['backgroundSize'] = ! empty( $background_styles['backgroundSize'] ) ? $background_styles['backgroundSize'] : 'cover';
-
-	if ( 'file' === $background_image_source && ! empty( $background_styles['backgroundImage']['url'] ) ) {
-		// If the background size is set to `contain` and no position is set, set the position to `center`.
-		if ( 'contain' === $background_styles['backgroundSize'] && ! isset( $background_styles['backgroundPosition'] ) ) {
-			$background_styles['backgroundPosition'] = 'center';
-		}
-	}
-
 	return gutenberg_style_engine_get_styles( array( 'background' => $background_styles ) );
```

Comment:
It looks like all this function does now is call `gutenberg_style_engine_get_styles`. Is it still worth keeping this function in? If we're not planning to add any other features to this function, then I think we could likely safely remove it since it hasn't been in Gutenberg very long. If we're thinking we might add more features to it as part of the background image UI in global styles, then please ignore this comment 😄

Reply:
>  then please ignore this comment

I just remembered that we'll likely still want a shared function for future functionality like retrieving an image based on a theme url, or potentially a featured image, so there's at least a couple of examples of potential uses for the function in the future 😄

Reply:
Very good point - I left it in because we might want some hackery for global styles as well.

I was hoping to know for sure when we flip the switch on block global styles in theme.json.

One other thing: the style engine handles both `"backgroundImage": "url(...)"` and `"backgroundImage":  { "url": "http://..." }`

But you're right. Apart from that it's not doing anything. 

I'm happy to cull it here, or keep it until we're sure we don't need it. 🤔

Reply:
Whichever way you want to go with it! I think it all depends on what abstract makes sense to a consumer. Calling "in" to a block support function from global styles, even if it only does one other thing isn't necessarily bad, as it helps abstract things a little, so conceptually I don't mind it.

Reply:
> One other thing: the style engine handles both "backgroundImage": "url(...)" and "backgroundImage":  { "url": "http://..." }

This is the only advantage as far as I can see.

Removing the function just means that theme.json users will have to use `"backgroundImage": "url(...)"` for now.

I don't think that's so bad, and we can always reinstate when things are more concrete.

Reply:
So I reverted it. 

Now we're nearly back to the state before:

- https://github.com/WordPress/gutenberg/pull/59354

😄